lib/rack/mock.rb

```diff
@@ -257,9 +257,13 @@ def identify_cookie_attributes(cookie_filling)
       cookie_bits.drop(1).each do |bit|
         if bit.include? '='
           cookie_attribute, attribute_value = bit.split('=', 2)
-          cookie_attributes.store(cookie_attribute.strip, attribute_value.strip)
           if cookie_attribute.include? 'max-age'
             cookie_attributes.store('expires', Time.now + attribute_value.strip.to_i)
+          elsif cookie_attribute.include? 'expires'
+            next if cookie_attributes.key? 'expires' # max-age attribute has precedence
+            cookie_attributes.store('expires', Time.httpdate(attribute_value.strip))
```

Comment:
I think this would be easier to understand if it used the more idiomatic form:

```ruby
cookie_attributes['expires'] ||= Time.httpdate(attribute_value.strip)
```

I realize that makes it less consistent with the rest of the method, but the rest of the method should probably switch from `store` to the more idiomatic `[]=`.